src/test/java/com/redhat/exhort/providers/Gradle_Provider_Test.java

```diff
@@ -0,0 +1,168 @@
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.mockStatic;
+
+@ExtendWith(HelperExtension.class)
+@ExtendWith(MockitoExtension.class)
+class Gradle_Provider_Test extends ExhortTest {
+//  private static System.Logger log = System.getLogger("Gradle_Provider_Test");
+  // test folder are located at src/test/resources/tst_manifests
+  // each folder should contain:
+  // - build.gradle: the target manifest for testing
+  // - expected_sbom.json: the SBOM expected to be provided
+  static Stream<String> testFolders() {
+    return Stream.of(
+      "deps_with_no_ignore_common_paths"
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("testFolders")
+  void test_the_provideStack(String testFolder) throws IOException, InterruptedException {
+    // create temp file hosting our sut build.gradle
+    var tmpGradleDir = Files.createTempDirectory("exhort_test_");
+    var tmpGradleFile = Files.createFile(tmpGradleDir.resolve("build.gradle"));
+//    log.log(System.Logger.Level.INFO,"the test folder is : " + testFolder);
+    try (var is = getClass().getClassLoader().getResourceAsStream(String.join("/","tst_manifests", "gradle", testFolder, "build.gradle"))) {
+      Files.write(tmpGradleFile, is.readAllBytes());
+    }
+    var settingsFile = Files.createFile(tmpGradleDir.resolve("settings.gradle"));
+    try (var is = getClass().getClassLoader().getResourceAsStream(String.join("/","tst_manifests", "gradle", testFolder, "settings.gradle"))) {
+      Files.write(settingsFile, is.readAllBytes());
+    }
+    var subGradleDir = Files.createDirectories(tmpGradleDir.resolve("gradle"));
+    var libsVersionFile = Files.createFile(subGradleDir.resolve("libs.versions.toml"));
+    try (var is = getClass().getClassLoader().getResourceAsStream(String.join("/","tst_manifests", "gradle", testFolder, "gradle", "libs.versions.toml"))) {
+      Files.write(libsVersionFile, is.readAllBytes());
+    }
+    // load expected SBOM
+    String expectedSbom;
+    try (var is = getClass().getClassLoader().getResourceAsStream(String.join("/","tst_manifests", "gradle", testFolder, "expected_stack_sbom.json"))) {
+      expectedSbom = new String(is.readAllBytes());
+    }
+    String depTree;
+    try (var is = getClass().getClassLoader().getResourceAsStream(String.join("/","tst_manifests", "gradle", testFolder, "depTree.txt"))) {
+      depTree = new String(is.readAllBytes());
+    }
+
+    MockedStatic<Operations> mockedOperations = mockStatic(Operations.class, Mockito.CALLS_REAL_METHODS);
+    ArgumentMatcher<Path> matchPath = path -> path == null;
+    mockedOperations.when(() -> Operations.runProcessGetOutput(argThat(matchPath),any(String[].class), any(String[].class))).thenReturn(depTree);
```

Comment:
@olavtar this is not working unfortunately , Mockito.CALLS_REAL_METHOD should be used only if you need to invoke at least one real static method from static class, this is not the case, as you need to mock everything, that is - both of gradle dependencies + gradle properties 
```suggestion
    String gradleProperties;
    try (var is = getClass().getClassLoader().getResourceAsStream(String.join("/","tst_manifests", "gradle", testFolder, "gradle.properties"))) {
      gradleProperties = new String(is.readAllBytes());
    }

    MockedStatic<Operations> mockedOperations = mockStatic(Operations.class);
    ArgumentMatcher<String> gradle = string -> string.equals("gradle");
    ArgumentMatcher<String> dependencies = string -> string.equals("dependencies");
    ArgumentMatcher<String> properties = string -> string.equals("properties");

    mockedOperations.when(() -> Operations.getCustomPathOrElse("gradle")).thenReturn("gradle");
    mockedOperations.when(() -> Operations.runProcessGetOutput(any(Path.class),  argThat(gradle),argThat(dependencies))).thenReturn(depTree);
    mockedOperations.when(() -> Operations.runProcessGetOutput(any(Path.class), argThat(gradle), argThat(properties))).thenReturn(gradleProperties);

```

Reply:
Done